templates/bin/fastcheck

```diff
@@ -2,6 +2,34 @@
 
 set -e
 
+# Adds a -c flag to only check files that have actually changed
+# e.g.: bin/fastcheck -c
+CHANGED_ONLY="$1"
+changed() {
+  if [ "$CHANGED_ONLY" = '-c' ]
+  then
+    CHANGES="$(git diff --name-only | sed -e "s@^@$(git rev-parse --show-cdup)@g" | grep -E "$1" || :)"
```

Comment:
can you explain me what are you doing with this line? That basically boils down to the whole logic behind this PR.

Reply:
The main flow goes something like this:
`git diff --name-only` - Get a list of changed filenames
`sed -e "s@^@$(git rev-parse --show-cdup)@g"` - replace the root path with the current workdir (`git rev-parse --show-cdup` returns the path from current dir all the way up to the git root) (the `@` in the sed command are an [alias](https://www.grymoire.com/Unix/Sed.html#uh-2) for `/` as we want to use paths inside our regex)
and `grep -E "$1"` just matches the files with the regex parameter $1